### DISCUSSION TITLE
OAuth2 三方登录时 AccessToken 应该如何颁发

### DISCUSSION THREAD

**Discussion by dingdayu** [vbenjs/vue-vben-admin#4653] [General]
这里不讨论 OAuth2 的授权流程，仅讨论授权完成后，如何为VBen 颁发 AccessToken .

比如点击 登录页面的 三方登录 Icon Button 后 openWindow 一个 URL 进行后端跳转，后端完成授权后会再次跳转到一个地方（通常跳转为 / 首页，同登录完成后的动作），请教这次跳转 如何颁发 Token？以接入现有的 账户密码登录的 Token。

**Comment by dingdayu** [comment]
在 `playground/src/api/request.ts` 的 createRequestClient 65行处，增加 获取 Cookie 中的 Token，如果 Store 中不存在，Cookie 中存在，则写入，如何？不知道与路由的权限判断，谁在前谁在后。

**Comment by dingdayu** [comment]
我查到的好像 是 使用隐藏式 授权，这种不是很保险，想要考虑 令牌式的授权，有什么建议吗？

**Reply by ArthurDarkstone** [reply]
可以考虑后端来做。在授权完成之后，后端提供接口将OAuth2的Token 将换为业务系统的token，接下来走正常的accessToken认证即可

**Reply by dingdayu** [reply]
感谢回复，目前的做法是：
1. 点击三方登录按钮后，通过 openWindow 进行页面跳转，
2. 后端（301）引导（并记录来访URL）拼接授权参数到三方授权页面，
3. 授权完成后再301跳转回一个后端地址，后端完成 Code 换 Token 并拉取个人资料，
4. 然后写登录凭证JWT to Cookie，然后跳转回前端页面（来访URL）。

**Comment by elarbor** [comment]
@dingdayu 对应的后端项目在哪啊？

**Reply by dingdayu** [reply]
内部项目，目前使用上面我提到的方法 通过 Cookie 颁发的方式，没有遇到特别的问题。
